include/deal.II/base/vectorization.h

```diff
@@ -229,12 +229,48 @@ class VectorizedArray
    * the amount of bytes in the vectorized array, as opposed to casting a
    * double address to VectorizedArray<double>*.
    */
-  void store (Number *ptr) const
+  DEAL_II_ALWAYS_INLINE void store (Number *ptr) const
   {
     *ptr = data;
   }
 
   /**
+   * Load @p n_array_elements from memory into the calling class, starting at
+   * the given address and with given offsets, each entry from the offset
+   * providing one element of the vectorized array.
+   *
+   * This operation corresponds to the following code (but uses a more
+   * efficient implementation in case the hardware allows for that):
+   * @code
+   * for (unsigned int v=0; v<VectorizedArray<Number>::n_array_elements; ++v)
+   *   this->operator[](v) = base_ptr[offsets[v]];
+   * @endcode
+   */
+  DEAL_II_ALWAYS_INLINE void gather (const Number *base_ptr,
+                                     const int    *offsets)
+  {
+    data = base_ptr[offsets[0]];
```

Comment:
In fact, for the default implementation of `VectorizedArray` (where no overload is present), we indeed only read a single element because `VectorizedArray<Number>` is essentially the same as `Number`. This is only different for specific Intel CPUs where `double` and `float` provide something else. Or did you mean something else?